src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp

```diff
@@ -148,7 +149,7 @@ void WaypointMovementGenerator<Creature>::ProcessWaypointArrival(Creature* creat
     // inform AI
     if (CreatureAI* AI = creature->AI())
     {
-        AI->MovementInform(WAYPOINT_MOTION_TYPE, _currentNode);
+        AI->MovementInform(WAYPOINT_MOTION_TYPE, waypoint.Id);
```

Comment:
this is a behavior change. check Blackrock Caverns Raz the Crazed behavior if all waypoint events are triggered properly